app/src/main/java/com/droidhats/campuscompass/views/MapFragment.kt

```diff
@@ -297,50 +319,40 @@ class MapFragment : Fragment(), OnMapReadyCallback, GoogleMap.OnMarkerClickListe
             bottomSheetBehavior.state = BottomSheetBehavior.STATE_EXPANDED
         }
 
-        //Checking which transportation mode is selected, default is walking.
-        var transportationMode: String = "driving"
-        var radioSelectedId = radioTransportGroup.checkedRadioButtonId
-        transportationMode = getTransportationMode(radioSelectedId)
-
-        //In case the transportation mode is changed, this will capture it.
-        radioTransportGroup.setOnCheckedChangeListener { _, optionId ->
-            transportationMode = getTransportationMode(optionId)
-        }
-
-        // Populate the bottom sheet with building information
-        val buildingName: TextView = activity!!.findViewById(R.id.bottom_sheet_building_name)
-        buildingName.text = p.tag.toString()
+        //Populate the bottom sheet with building information
+        val buildingNameText: TextView = requireActivity().findViewById(R.id.bottom_sheet_building_name)
+        buildingNameText.text = p.tag.toString()
 
+        //Navigation here
         val directionsButton: Button = requireActivity().findViewById(R.id.bottom_sheet_directions_button)
         directionsButton.setOnClickListener(View.OnClickListener {
-
-            ///TODO: Refactor this, no longer needed since buildings.json holds the location of the building @Makram
```

Comment:
So I mentioned this in another branch, but I really don't think that we should be refactoring this. This code snippet is highly dependant on the polygons and buildings having the same name, and the polygon's tag/name is only set by a line down below (meaning we are not going to remember this dependency on that line later on). In addition, I really don't think it will take any hit to performance, since it's not really a big calculation, so I would prefer not accepting this change.